### PR TITLE
Restructuring the adapters to use requireNonNull & single returns

### PR DESCRIPTION
**Pull request by viktorklang**
No semantical changes. :)

### CODE REVIEW

File: api/src/main/java9/org/reactivestreams/FlowAdapters.java [hunk 1]
```diff
@@ -157,12 +158,12 @@ private FlowAdapters() {
     static final class FlowToReactiveSubscription implements Flow.Subscription {
         final org.reactivestreams.Subscription reactiveStreams;
 
-        public FlowToReactiveSubscription(org.reactivestreams.Subscription reactive) {
+        public FlowToReactiveSubscription(final org.reactivestreams.Subscription reactive) {
```
Comment (ktoso) [inline api/src/main/java9/org/reactivestreams/FlowAdapters.java]: Other than personal preference, in classes you PRed, we don't really do this style wise in this project...

File: api/src/main/java9/org/reactivestreams/FlowAdapters.java [hunk 2]
```diff
@@ -31,16 +32,16 @@ private FlowAdapters() {
     @SuppressWarnings("unchecked")
     public static <T> org.reactivestreams.Publisher<T> toPublisher(
             Flow.Publisher<? extends T> flowPublisher) {
-        if (flowPublisher == null) {
-            throw new NullPointerException("flowPublisher");
-        }
+        requireNonNull(flowPublisher, "flowPublisher");
```
Comment (ktoso) [inline api/src/main/java9/org/reactivestreams/FlowAdapters.java]: ok

Comment (viktorklang) [reply api/src/main/java9/org/reactivestreams/FlowAdapters.java]: :)

**Comment by viktorklang** [comment]
@ktoso final params removed

**Comment by viktorklang** [comment]
@reactive-streams/contributors Merging

**Pull request merged**